src/ray/raylet/node_manager.cc

```diff
@@ -1933,79 +1933,127 @@ void NodeManager::FinishAssignedActorTask(Worker &worker, const Task &task) {
-            PublishActorStateTransition(
-                actor_id, new_actor_data,
-                /*failure_callback=*/
-                [](gcs::AsyncGcsClient *client, const ActorID &id,
-                   const ActorTableData &data) {
-                  // Only one node at a time should succeed at creating the actor.
-                  RAY_LOG(FATAL) << "Failed to update state to ALIVE for actor " << id;
-                });
-          },
-          [actor_id](ray::gcs::AsyncGcsClient *client, const UniqueID &checkpoint_id) {
-            RAY_LOG(FATAL) << "Couldn't find checkpoint " << checkpoint_id
-                           << " for actor " << actor_id << " in GCS.";
-          }));
-    } else {
-      // The actor did not resume from a checkpoint. Immediately notify the
-      // other node managers that the actor has been created.
-      HandleActorStateTransition(actor_id, ActorRegistration(new_actor_data));
-      PublishActorStateTransition(
-          actor_id, new_actor_data,
-          /*failure_callback=*/
-          [](gcs::AsyncGcsClient *client, const ActorID &id, const ActorTableData &data) {
-            // Only one node at a time should succeed at creating the actor.
-            RAY_LOG(FATAL) << "Failed to update state to ALIVE for actor " << id;
-          });
-    }
+    auto new_actor_data = CreateActorTableDataFromCreationTask(task);
+    // Lookup the parent actor id
+    auto parent_task_id = task.GetTaskSpecification().ParentTaskId();
+    RAY_CHECK_OK(gcs_client_->raylet_task_table().Lookup(
+        JobID::Nil(), parent_task_id,
+        //success_callback
+        [this, task, actor_id, actor_handle_id, new_actor_data, resumed_from_checkpoint]
+           (ray::gcs::AsyncGcsClient *client, const TaskID &parent_task_id,
+               const TaskTableData &parent_task_data) mutable {
+          // The task was in the GCS task table. Use the stored task spec to
+          // get the parent actor id.
+          auto message = flatbuffers::GetRoot<protocol::Task>(parent_task_data.task().data());
+          Task parent_task(*message);
+          new_actor_data.set_parent_actor_id(parent_task.GetTaskSpecification().ActorId().Binary());
+          FinishAssignedActorTaskHelper(actor_id, new_actor_data, resumed_from_checkpoint);
+          if(!resumed_from_checkpoint){
+            // The actor was not resumed from a checkpoint. We extend the actor's
+            // frontier as usual since there is no frontier to restore.
+            ExtendActorFrontier(task, actor_id, actor_handle_id);
+          }
+        },
+        //failure_callback
+        [this, task, actor_id, actor_handle_id, new_actor_data, resumed_from_checkpoint]
+            (ray::gcs::AsyncGcsClient *client, const TaskID &parent_task_id) mutable {
+          // The parent task was not in the GCS task table. It must therefore be in the
+          // lineage cache.
+          RAY_CHECK(lineage_cache_.ContainsTask(parent_task_id))
```

Comment:
Unfortunately, one of the travis tests for the new C++ worker is failing on this check, I think because it does not add the driver task to the GCS. Let's make this a non-fatal check for now. If the task is not in the GCS or the lineage cache, we can just set the parent actor ID to nil (and leave a note/todo explaining this case).

Reply:
Done.